joblib/pool.py

```diff
@@ -326,6 +327,8 @@ def terminate(self):
         n_retries = 10
         for i in range(n_retries):
             try:
+                if os.name != "nt":
+                    self._inqueue._reader.close()
```

Comment:
Probably we don't need this anymore after removing support for return_generator in multiprocessing